### PR TITLE
feat: struct name camelcase

### PR DESCRIPTION
**Pull request by 0xRaccoon**

### CODE REVIEW

File: rules/struct-name-camelcase.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    category: 'Style Guide Rules',
+  },
+
+  isDefault: true,
+  recommended: true,
+  defaultSetup: 'warn',
+  schema: null,
+}
+
+class StructNameCamelcaseChecker extends BaseChecker {
```
Comment (gotzenx) [inline rules/struct-name-camelcase.js]:
```suggestion
class StructNameCamelCaseChecker extends BaseChecker {
```

File: test/rules/struct-name-camelcase.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    rules: {'struct-name-camelcase': 'error'},
+};
+
+describe('Linter - struct-name-camelcase', () => {
+    it('should raise error for struct name not in camelcase', () => {
+        const code = contractWith('struct teststruct {}');
+        const report = linter.processStr(code, config);
+
+        assert.equal(report.errorCount, 1);
+        assert.ok(report.messages[0].message == `Struct name 'teststruct' must be in CamelCase`);
```
Comment (gotzenx) [inline test/rules/struct-name-camelcase.ts]: I would suggest to use a variable like `structName = 'teststruct'`. So then that variable is used to:
1. create the code variable
2. assert.ok
That way we ensure that lines of code will always use the same struct name

Comment (gotzenx) [reply test/rules/struct-name-camelcase.ts:15]:
```suggestion
        const structName = 'teststruct';
        const code = contractWith(`struct ${structName} {}`);
        const report = linter.processStr(code, config);

        assert.equal(report.errorCount, 1);
        assert.ok(report.messages[0].message == `Struct name ${structName} must be in CamelCase`);
```

Comment (gotzenx) [reply test/rules/struct-name-camelcase.ts]:
```
        const structName = 'teststruct';
        const code = contractWith(`struct ${structName} {}`);
        const report = linter.processStr(code, config);

        assert.equal(report.errorCount, 1);
        assert.ok(report.messages[0].message == `Struct name ${structName} must be in CamelCase`);
```

File: test/utils/checkers.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import configObject from 'solhint/lib/config';
 import { validSeverityMap } from 'solhint/lib/config/config-validator';
 import NamedReturnValuesChecker from '../../rules/named-return-values';
 import NonStateVarsLeadingUnderscoreChecker from '../../rules/non-state-vars-leading-underscore';
+import StructNameCamelcaseChecker from '../../rules/struct-name-camelcase';
```
Comment (gotzenx) [inline test/utils/checkers.ts]:
```suggestion
import StructNameCamelCaseChecker from '../../rules/struct-name-camelcase';
```

**Review by gotzenx — APPROVED** [review]

**Review by 0xGorilla — APPROVED** [review]

**Pull request merged**